pycoal/mineral.py

```diff
@@ -362,3 +368,184 @@ def convert(cls, data_dir="", db_file="", hdr_file=""):
+            an `ASTER Spectral Library Version 2.0 <https://asterweb.jpl.nasa.gov/>`_ .spectrum.txt file
+            ASTER Library Version 2.0 Spectral Library files are in .spectrum.txt file format
+            
+            Spectral Library Version 7 can be downloaded `here <https://speclab.cr.usgs.gov/spectral-lib.html>`_
+            
+            Args:
+            library_filename (str): path to Spectral File you wish to convert
+            """
+        if not library_filename:
+            raise ValueError("Must provide path for Spectral File.")
+        
+        line_count = 1
+        with open(library_filename,'r') as input_file:
+            for line_count, l in enumerate(input_file):
+                pass
+            
+        input_file = open(library_filename,'r')
+        #Read Name of Spectra on first line of the file
+        spectra_line = input_file.readline()
+        spectra_name = spectra_line[23:]
+        k = 0
+        #Loop through file and store all wavelength values for the given Spectra
+        spectra_values_file = open('SpectraValues.txt','w')
+        spectra_wave_length = 0
+        while(k < line_count):
+            spectra_wave_length = float(input_file.readline()) * 100
+            spectra_wave_length = spectra_wave_length / 1000
+            spectra_wave_length = float("{0:.5f}".format(spectra_wave_length))
+            spectra_y_value = spectra_wave_length * 10
+            line = str(spectra_wave_length) + '  ' + str(spectra_y_value)
+            spectra_values_file.write(line)
+            spectra_values_file.write('\n')
+            k = k+1
+        #Write new file in the form of an ASTER .spectrum.txt file while using stored
+        #Spectra Name and stored Spectra Wavelength values`
+        input_file = open(library_filename,'w')
+        input_file.write('Name:')
+        input_file.write(spectra_name)
+        input_file.write('Type:\n')
+        input_file.write('Class:\n')
+        input_file.write('Subclass:\n')
+        input_file.write('Particle Size:  Unknown\n')
+        input_file.write('Sample No.:  0000000000\n')
+        input_file.write('Owner:\n')
+        input_file.write('Wavelength Range:  ALL\n')
+        input_file.write('Origin: Spectra obtained from the Noncoventional Exploitation Factors\n')
+        input_file.write('Data System of the National Photographic Interpretation Center.\n')
+        input_file.write('Description:  Gray and black construction asphalt.  The sample was\n')
+        input_file.write('soiled and weathered, with some limestone and quartz aggregate\n')
+        input_file.write('showing.\n')
+        input_file.write('\n')
+        input_file.write('\n')
+        input_file.write('\n')
+        input_file.write('Measurement:  Unknown\n')
+        input_file.write('First Column:  X\n')
+        input_file.write('Second Column: Y\n')
+        input_file.write('X Units:  Wavelength (micrometers)\n')
```

Comment:
Writing all of these values still seems way too tightly tied with some very specific file. It is not correct and needs to be much more flexible.